eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -802,6 +803,38 @@ object Helpers {
         }).toSet.flatten
       }
 
+    /**
+      * Tells if we were the origin of this outgoing htlc
+      *
+      * @param htlcId
+      * @param originChannels
+      * @return
+      */
+    def isSentByLocal(htlcId: Long, originChannels: Map[Long, Origin]) = (originChannels.get(htlcId) collect { case l: Local => l }).isDefined
```

Comment:
I wonder if there is a cleaner way to do that

Reply:
Perhaps this is a bit more readable
```suggestion
    def isSentByLocal(htlcId: Long, originChannels: Map[Long, Origin]) = originChannels.get(htlcId) match {
      case Some(Local(_)) => true
      case _ => false
    }
```